photon-api/src/main/scala/com/linkedin/photon/ml/data/RandomEffectDataSet.scala

```diff
@@ -295,7 +297,13 @@ object RandomEffectDataSet {
       gameDataSet: RDD[(UniqueSampleId, GameDatum)],
       randomEffectDataConfiguration: RandomEffectDataConfiguration,
       randomEffectPartitioner: Partitioner,
-      existingModelKeysRddOpt: Option[RDD[REId]]): RDD[(REId, LocalDataSet)] = {
+      existingModelKeysRddOpt: Option[RDD[REId]]): (RDD[(REId, LocalDataSet)], Array[Double]) = {
+
+    val ungroupedFeatures = gameDataSet
+      .values
+      .filter( _.response == 1D)
+      .map( _.featureShardContainer(randomEffectDataConfiguration.featureShardId)).reduce(_ + _)
```

Comment:
can you name some examples? I have given it some thoughts. I think they are equivalent in our cases since the columns are independent and we are evaluating counts instead of probability. So this form actually facilitate our computation.

Reply:
You're right -- the counts are equivalent. Sorry for the confusion.

Reply:
Example samples:
```
| y | f_1 | f_2 |
|---|-----|-----|
| 1 |  1  |  0  |
| 1 |  1  |  1  |
| 1 |  0  |  1  |
| 0 |  1  |  0  |
| 0 |  1  |  1  |

Ratio (f_1 == 1 | y == 1) = 2/3
Ratio (y == 1 | f_1 == 1) = 1/2
```

Reply:
i am only evaluating the counts here.
count(y|f_1)=count(f_1|y)=2